test/src/test/java/org/corfudb/integration/AbstractIT.java

```diff
@@ -657,29 +703,14 @@ public Process runServer() throws IOException {
             }
             ProcessBuilder builder = new ProcessBuilder();
 
-            if (!metricsConfigFile.isEmpty()) {
-                addMetricsToProcessBuilder(builder, "corfu_replication_server");
-            } else {
-                builder.command("sh", "-c", "bin/corfu_replication_server " + getOptionsString());
-            }
+            builder.command(SH, HYPHEN_C, getCodeCoverageCmd() + getMetricsCmd(metricsConfigFile) +
+                    " bin/corfu_replication_server " + getOptionsString());
 
             builder.directory(new File(CORFU_PROJECT_DIR));
             Process corfuReplicationServerProcess = builder.start();
             StreamGobbler streamGobbler = new StreamGobbler(corfuReplicationServerProcess.getInputStream(), serverConsoleLogPath);
             Executors.newSingleThreadExecutor().submit(streamGobbler);
             return corfuReplicationServerProcess;
         }
-
-        private void addMetricsToProcessBuilder(ProcessBuilder builder, String runnable) throws IOException {
```

Comment:
No, this was made by me. I refactored Pavel's code since that was also exporting a variable value and invoking the shell script.
Basically, I made both invocations similar and simpler, and it can be further extended later on if there are more such invocations.